/* Generated by nanopb-0.3.9.3 at Tue Oct 08 20:41:38 2019. */

syntax = "proto3";

service Webserver {
    rpc put_message(server_req) returns (vessel_state) {}
}

service Teensyserver {
    rpc put_message(vessel_state) returns (vessel_state) {}
}

enum TRIM_STATE {
    MIN_LIFT = 0;
    STBD_TACK = 1;
    PORT_TACK = 2;
    MAX_DRAG_STBD = 3;
    MAX_DRAG_PORT = 4;
    MAN_CTRL = 5;
}

enum DEVICE_ID {
    RIGID_SAIL = 0;
    HERO = 1;
    ARDUINO = 2;
    BBB = 3;
}

message vessel_state {
    DEVICE_ID device_id = 1;
    TRIM_STATE state = 2;
    uint32 curHeelAngle = 3;
    uint32 maxHeelAngle = 4;
    uint32 controlAngle = 5;
    uint32 windAngle = 6;
    uint32 vIn = 7;
    bool hallPortTrip = 8;
    bool hallStbdTrip = 9; 
}

message server_req{
    bool succ = 1;
}